src/i965_drv_video.c

```diff
@@ -1483,7 +1483,12 @@ i965_suface_external_memory(VADriverContextP ctx,
         index >= memory_attibute->num_buffers)
         return VA_STATUS_ERROR_INVALID_PARAMETER;
 
-    obj_surface->size = memory_attibute->data_size;
+    if (!memory_attibute->data_size)
+        // no data_size provided by client, calculate it here
+        obj_surface->size = memory_attibute->height * memory_attibute->pitches[0];
```

Comment:
This is wrong. You should consider the pitches of all planes in order to calculate the obj_surface size.
Please have a look into the native_memory type implementation where the obj_surface size calculated correctly.